### PR TITLE
tests: Remove unused testing code

### PR DESCRIPTION
**Pull request by practicalswift**
Remove unused testing code.

Rationale:
* Less is more :-)
* Unused code is by definition "untested"
* [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

### CODE REVIEW

File: test/functional/test_framework/test_framework.py [hunk 1]
```diff
@@ -515,18 +515,6 @@ def skip_if_no_wallet(self):
         if not self.is_wallet_compiled():
             raise SkipTest("wallet has not been compiled.")
 
-    def skip_if_no_cli(self):
```
Comment (MarcoFalke) [inline test/functional/test_framework/test_framework.py]: Should probably call this:

```diff
diff --git a/test/functional/test_framework/test_framework.py b/test/functional/test_framework/test_framework.py
index 9a589240a8..238b50387d 100755
--- a/test/functional/test_framework/test_framework.py
+++ b/test/functional/test_framework/test_framework.py
@@ -161,8 +161,10 @@ class BitcoinTestFramework(metaclass=BitcoinTestMetaClass):
         success = TestStatus.FAILED
 
         try:
-            if self.options.usecli and not self.supports_cli:
-                raise SkipTest("--usecli specified but test does not support using CLI")
+            if self.options.usecli:
+                if not self.supports_cli:
+                    raise SkipTest("--usecli specified but test does not support using CLI")
+                self.skip_if_no_cli()
             self.skip_test_if_missing_module()
             self.setup_chain()
             self.setup_network()

File: test/functional/test_framework/mininode.py [hunk 2]
```diff
@@ -313,7 +313,7 @@ def on_ping(self, message):
         self.send_message(msg_pong(message.nonce))
 
     def on_verack(self, message):
-        self.verack_received = True
+        pass
```
Comment (MarcoFalke) [inline test/functional/test_framework/mininode.py]: ack

File: test/functional/test_framework/messages.py [hunk 3]
```diff
@@ -1074,19 +1070,6 @@ def serialize(self):
...
-class msg_generic():
-    def __init__(self, command, data=None):
-        self.command = command
-        self.data = data
-
-    def serialize(self):
-        return self.data
-
-    def __repr__(self):
-        return "msg_generic()"
```
Comment (MarcoFalke) [inline test/functional/test_framework/messages.py]: ack

File: test/functional/test_framework/messages.py [hunk 4]
```diff
@@ -54,9 +54,6 @@
 def sha256(s):
     return hashlib.new('sha256', s).digest()
 
-def ripemd160(s):
-    return hashlib.new('ripemd160', s).digest()
```
Comment (MarcoFalke) [inline test/functional/test_framework/messages.py]: When has this last been used? Not sure if we want to remove library functions that are still used in other branches. Same for the key.py changes.

File: test/functional/feature_cltv.py [hunk 5]
```diff
@@ -25,7 +25,7 @@
 
 # Reject codes that we might receive in this test
 REJECT_INVALID = 16
-REJECT_OBSOLETE = 17
+# REJECT_OBSOLETE = 17
```
Comment (MarcoFalke) [inline test/functional/feature_cltv.py]: When was the first and last use? Might want to remove altogether after investigating that.

Comment (practicalswift) [reply test/functional/feature_cltv.py]: Last use was removed in your commit fac3e22b18cd29053bc17065fd75db7b84ba6f40 :-)

File: test/functional/test_framework/messages.py (hunk 4 above)
Comment (practicalswift) [reply test/functional/test_framework/messages.py]: Last use was removed in b9f34e84befa7db6ff8c9b92a09d0dfa40388fb7

Comment (MarcoFalke) [reply test/functional/test_framework/messages.py]: ack

File: test/functional/feature_cltv.py (hunk 5 above)
Comment (MarcoFalke) [reply test/functional/feature_cltv.py]: So please remove the constants here as well.

**Comment by practicalswift** [comment]
@MarcoFalke Thanks for the quick review. Updated. Please re-review :-)

File: test/functional/test_framework/messages.py [hunk 6]
```diff
@@ -31,7 +31,7 @@
 MY_SUBVERSION = b"/python-mininode-tester:0.0.3/"
 MY_RELAY = 1 # from version 70001 onwards, fRelay should be appended to version messages (BIP37)
 
-MAX_INV_SZ = 50000
+# MAX_INV_SZ = 50000
```
Comment (MarcoFalke) [inline test/functional/test_framework/messages.py]: remove? (This is no longer used after the removal of the comptool)

Comment (practicalswift) [reply test/functional/test_framework/messages.py]: Done!

File: test/functional/test_framework/test_framework.py [hunk 7]
```diff
@@ -161,8 +161,10 @@ def main(self):
         success = TestStatus.FAILED
 
         try:
-            if self.options.usecli and not self.supports_cli:
-                raise SkipTest("--usecli specified but test does not support using CLI")
+            if self.options.usecli:
+                if not self.supports_cli:
+                    raise SkipTest("--usecli specified but test does not support using CLI")
+                self.skip_if_no_cli()
```
Comment (promag) [inline test/functional/test_framework/test_framework.py]: Does this change belong here?

File: test/functional/test_framework/messages.py [hunk 8]
```diff
@@ -1136,7 +1131,6 @@ def serialize(self):
     def __repr__(self):
         return "msg_block(block=%s)" % (repr(self.block))
 
-
```
Comment (promag) [inline test/functional/test_framework/messages.py]: nit, unrelated.

**Review by promag — COMMENTED** [review]
Concept ACK, agree with rationale.

Mind sharing how you find these unused code?

**Comment by practicalswift** [comment]
@promag `vulture` is a great tool for finding dead Python code. See #14365 for `vulture` integration in Travis :-)

**Comment by practicalswift** [comment]
Closing this PR in favour of #14365 as suggested by @promag :-)
